backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -158,13 +177,27 @@ private void initializeS3ClientWithTemporaryCredentials(Regions clientRegion) {
 
   private void RefreshCredentialsAndSchedule(String region) {
     LOGGER.debug("fetching token for s3 access");
+    if (task != null) {
```

Comment:
comment for conrado's review: if there is a scheduled task cancel it since this refresh is off-cycle and will start a refresh cycle of its own